check.py

```diff
@@ -192,7 +192,11 @@ def run_spec_tests():
     print('\n[ checking wasm-shell spec testcases... ]\n')
 
     for wast in shared.options.spec_tests:
-        print('..', os.path.basename(wast))
+        base = os.path.basename(wast)
+        print('..', base)
+        # windows has some failures that need to be investigated
+        if base == 'names.wast' and shared.skip_if_on_windows('spec: ' + base):
```

Comment:
Why skip `names.wast`? If this is not something windows-related, we have something similar in shared.py:
https://github.com/WebAssembly/binaryen/blob/454a1cb0b90397593472b9dd31c5418e23c40be3/scripts/test/shared.py#L385-L440

Reply:
Hmm, maybe it should be added to that list then, good point. It's windows-specific but we could append it when on that platform. However, then we can't use the `skip_if_on_windows` function which otherwise would be a single easy way to find all skipped stuff in windows... what's better?

Reply:
Oh, sorry, I didn't see `names.wast` is skipped _only_ on Windows.. I thought it was an 'or' condition, so I thought the condition was if it is `names.wast` or it is on Windows. [This part](https://github.com/WebAssembly/binaryen/blob/454a1cb0b90397593472b9dd31c5418e23c40be3/scripts/test/shared.py#L439-L440) of the code excludes failing tests from the list of spec tests, so if we add some platform handling logic I think we can make those files be excluded from here. But if we have only a single case of `names.wast` that is platform-dependent, I'm not sure that's worth it. Also this list is only for spec tests, so maybe it'd be better to keep the separate after all, unless we make a more complex system of skipping tests that takes the platform and the kind of tests as arguments, which seems like an overkill.

Reply:
Seems like overkill to me too, to try to generalize this. I don't feel strongly, but I think I prefer the current status of the code.